### DISCUSSION TITLE
[2024-05-27] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#10080] [Weekly Demo Day]
- James: Allow a Table to be used to `rename_columns`. #9940
- James: Show ALIAS matches as <name> (<alias>) #9955
- James: Update ICONs, linting fixes and add select_columns ALIASes. #9985
- James: Some fixes to make linter happy #10004
- James: Align headers in doc comments... #10025
- James: Some tweaks following Steve's testings #10042
- Radosław: Cloud integration:
	- Creating datalinks from code #9957
	- Prepare for the Path resolver change to GET #9958
	- Cloud Integration updates: renames in file metadata structure, remove path resolver workaround, partial fix for datalink resolver issue #10050
	- Investigated and summarized path resolution issues in Cloud https://github.com/enso-org/cloud-v2/issues/1252
- Radosław: Improvements to `Table.union` #9968
- Greg: `BigDecimalBuilder` and arithmetic operations. #9950
- Greg: Add `no_warning_propagation` flag to `Vector.build` and `Builder.new` #10079
- Ad: Make parse_fast_json private #9978
- Ad: Update test framework to default to only show failures #9986
- Ad: Add named_pattern #10015
- Ad: Added bold-italic-strikethrough icons #10030
- Ad: Add copy2 icon #10036
- Ad: Add named_pattern and make it usable in Text_Replace #10040
- Ad: Fix input_number icons #10066
- Ad: Add expression alias #10067
- Ad: Add new icons #10070
- Martha: Format Numbers in tables #10017
- Jaroslav: Reverse order of stack frames and test the result #9954
- Jaroslav: Table of `Persistance.Reference` at the end of the stream #9972
- Jaroslav: Use proper Java class name in error message #9996
- Jaroslav: Register all polyglot java import classes for refective access #9997
- Jaroslav: Include milliseconds in the log output #10016
- Jaroslav: Reactive `RuntimeCache` #10065
- Hubert: Abort jobs as early as possible #9927
- Hubert: Follow up to #9927 #9930
- Hubert: Use simple type name as Jackson type id #9961
- Hubert: Reduce ops when setting exisitng execution env #9967
- Hubert: Add an option to profile AppImage #9998
- Hubert: Experiment with replacing Jackson serde #10035
- Hubert: Remove invalid JVM parameters #10053
- Hubert: Turn strict errors off for Language Server #10061
- Hubert: `--attach-debugger` to start Language Server with debugger from project manager #10062
- Pavel: Ensure disable private check cmd line option disables runtime private access checks #10034
- Pavel: Improve performance of import export resolution #10043
- Pavel: Implement private methods #10060
- Dmitry: Allow file/read to return the contents of a collaborative buffer #9994
- Dmitry: Filter private constructors from Suggestions Database #10032
- Dmitry: Removing last node from main graph makes execution error #10048
- Dmitry: Increase setInterval delay in Ydoc test #10055
- Dmitry: Local dashboard shouldn't be showing system files #10072

**Comment by JaroslavTulach** [comment]
#### Reactive Cache

[ReactiveCache.webm](https://github.com/enso-org/enso/assets/26887752/b00b7eb2-02cb-4042-8f23-e7f51a988a25)

Accessing the arguments of `as_list` function by using `cache` to read value of `a` and `b` argument. The video shows the dropdown items are updated whenever the argument changes:

**Comment by radeusgd** [comment]
Now Union is more lenient on types - it allows mixing by default - but the less likely combinations get a relevant warning:

![image](https://github.com/enso-org/enso/assets/1436948/fed4920f-16f3-48f4-b350-25facce54b73)
